CMakeLists.txt

```diff
@@ -0,0 +1,63 @@
+cmake_minimum_required (VERSION 2.8.12)
+project (notepadqq)
+
+# Automation for QT 
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+set(CMAKE_AUTOMOC ON)
+set(CMAKE_AUTORCC ON)
+set(CMAKE_AUTOUIC ON)
+
+# Set our output directories
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+
+#Find QT packages required for build
+find_package(Qt5Core REQUIRED)
+find_package(Qt5Gui REQUIRED)
+find_package(Qt5Svg REQUIRED)
+find_package(Qt5Widgets REQUIRED)
+find_package(Qt5WebKitWidgets REQUIRED)
+find_package(Qt5PrintSupport REQUIRED)
+find_package(Qt5Network REQUIRED)
+find_package(Qt5LinguistTools REQUIRED)
+if(Qt5Core_FOUND)
+    message("Qt5Core found: Version ${Qt5Core_VERSION_STRING}")
+endif()
+
+# Initialize git submodules if git is available
+find_package(Git)
+if(GIT_FOUND)
+  message("git found: ${GIT_EXECUTABLE}")
+  if(EXISTS "${CMAKE_SOURCE_DIR}/.git")
+      message("Initializing submodules.")
+      execute_process(COMMAND "${GIT_EXECUTABLE}" submodule init)
+      execute_process(COMMAND "${GIT_EXECUTABLE}" submodule update)
+  endif()
+endif()
+
+
+# Installation instructions
```

Comment:
You shouldn't hardcode things like that. This will make it hard for distributors to package notepadqq correctly. Especially things like `install(TARGETS ${PROJECT_NAME} DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/${PROJECT_NAME})` further down.

Instead, use [GNUInstallDirs](https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html)